dijkstra3d.hpp

```diff
@@ -44,6 +44,13 @@ inline float _c(const float wa, const float wb, const float wc) {
   return std::sqrt(wa * wa + wb * wb + wc * wc);
 }
 
+// helper function to compute cross product
+inline float _cross(const float sx, const float sy, const float sz, 
+const float tx, const float ty, const float tz) {
+  return std::sqrt((sy*tz - sz*ty) * (sy*tz - sz*ty) + (sz*tx - sx*tz) * (sz*tx - sx*tz) + (sx*ty - sy*tx) * (sx*ty - sy*tx) / 
+          std::sqrt(sx*sx + sy*sy + sz*sz) / std::sqrt(tx*tx + ty*ty + tz*tz));
+}
```

Comment:
I think this normalization is only valid if the vectors are orthogonal (i.e. sine of the angle between them is 1), otherwise it is an overestimate.

Reply:
What `_cross` returns is `|sin(theta)|`.

This heuristic factor at [line 1270](https://github.com/seung-lab/dijkstra3d/pull/41/files/fbf41ac5a51258ea1f1bbcbf0f7a08bf61a17caa#diff-2ff96b0927bc0681208d8910c7e2858c11f4d39294b38d3524dad1984d1bd66fR1270) penalizes the direction that  doesn't align with vector pointing from source to target as have been suggested [here](http://theory.stanford.edu/~amitp/GameProgramming/Heuristics.html#breaking-ties). 


Some mathematical clarification:

Let 
```math
\vec{s}=(s_x, s_y, s_z), \vec{t}=(t_x, t_y, t_z)
```
then [cross product](https://en.wikipedia.org/wiki/Cross_product#Mnemonic) of these vectors are,
```math
\vec{s} \times \vec{t} = (s_yt_z - s_zt_y, s_zt_x - s_zt_z, s_xt_y - s_yt_x)
```
and since
```math
\| \vec{s} \times \vec{t} \| = \| \vec{s} \| \|\vec{t}\| | \sin <\vec{s}, \vec{t}> |
```

as well as
```math
\begin{split}
||\vec{s}|| = \sqrt{s_x^2 + s_y^2 + s_z^2} \
||\vec{t}|| = \sqrt{t_x^2 + t_y^2 + t_z^2} \
||\vec{s} \times \vec{t} || = \sqrt{ (s_yt_z - s_zt_y)^2 + (s_zt_x - s_zt_z)^2 + (s_xt_y - s_yt_x)^2 } \
\end{split}
```
So
```math

 | \sin <\vec{s}, \vec{t}> | =  \frac{\sqrt{ (s_yt_z - s_zt_y)^2 + (s_zt_x - s_zt_z)^2 + (s_xt_y - s_yt_x)^2 }} {\sqrt{s_x^2 + s_y^2 + s_z^2}  \sqrt{t_x^2 + t_y^2 + t_z^2} }
```

Reply:
Ah, okay. I think I understand what you're doing now.